api_tests/lib/actor.ts

```diff
@@ -106,17 +106,46 @@ export class Actor {
         if (network != "regtest") {
             throw Error("Expected network regtest, found " + network);
         }
-        let type = action.type;
-
-        switch (type) {
+        switch (action.type) {
             case "bitcoin-send-amount-to-address": {
                 action.payload.should.include.all.keys("to", "amount");
                 let { to, amount } = action.payload;
 
                 return this.wallet.btc().sendToAddress(to, parseInt(amount));
             }
             case "bitcoin-broadcast-signed-transaction": {
-                let { hex } = action.payload;
+                action.payload.should.include.all.keys("hex", "locktime");
+
+                let fetchMedianTime = async () => {
+                    let blockchainInfo = await bitcoin.getBlockchainInfo();
+                    return blockchainInfo.mediantime;
+                };
+
+                let { hex, min_median_time } = action.payload;
+
+                if (min_median_time) {
+                    let medianTime = await fetchMedianTime();
+                    let diff = min_median_time - medianTime;
```

Comment:
🔧 maybe worth doing the diff calculation directly in a function that also fetches the median time.

Reply:
Then I have to return two values, which is kind of awkward in JavaScript. I am aware of the duplication, I just preferred being explicit over making it too complicated.

Reply:
why 2 values? just return the diff.

Reply:
Need the median time for logging, unless I do it in the function.

Reply:
🙃 I was against the logging